lib/rspec-puppet/adapters.rb

```diff
@@ -113,12 +113,18 @@ class Adapter40 < Base
       #
       # @api private
       #
-      # Set the FacterImpl constant to the given Facter implementation.
-      # The method noops if the constant is already set
+      # Set the FacterImpl constant to the given Facter implementation or noop
+      # if the constant is already set.
+      #
+      # The method can also accept a proc parameter that will be evaluated if
+      # FacterImpl is not defined.
```

Comment:
Should this be in `@param`?

Reply:
Done! I detailed the documentation a bit, and added a description to `@param`.